src/cxx/basic_api.cc

```diff
@@ -468,6 +468,32 @@ JNIEXPORT void JNICALL Java_foundation_pEp_jniadapter_Engine__1config_1passphras
     }
 }
 
+JNIEXPORT void JNICALL Java_foundation_pEp_jniadapter_Engine__1config_1passphrase_1for_1new_1keys_1by_1email(JNIEnv *env,
+                                                                                                  jobject obj,
+                                                                                                  jboolean enable,
+                                                                                                  jbyteArray email,
+                                                                                                  jbyteArray passphrase)
+{
+    std::mutex *mutex_local = nullptr;
+    {
+        std::lock_guard<std::mutex> l(global_mutex);
+        pEpLog("called with lock_guard");
+        mutex_local = get_engine_java_object_mutex(env, obj);
+    }
+    std::lock_guard<std::mutex> l(*mutex_local);
+
+    bool _enable = static_cast<bool>(enable);
+    const char *_passphrase = to_string(env, passphrase);
+    const char *_email = to_string(env, email);
+
+    if (_enable) {
+        passphrase_cache.add(_email, _passphrase);
+    }
+    else {
+        passphrase_cache.remove(_passphrase);
```

Comment:
That looks wrong to me. The passphrase_cache method here expects the account email, you provide the passphrase.

Reply:
Thanks for that, updated.